.github/workflows/codeql-analysis.yml

```diff
@@ -17,10 +17,12 @@ jobs:
       # Initializes the CodeQL tools for scanning.
       - name: Initialize CodeQL
         uses: github/codeql-action/init@v1
+        if: ${{!github.event.repository.private}}
```

Comment:
You can put the `if` [directly on the job](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idif).

Reply:
Ah!  Thanks for the pointer.